econplayground/main/views.py

```diff
@@ -60,6 +64,81 @@ def test_func(self):
         return user_is_instructor(self.request.user)
 
 
+class GraphCloneDisplay(LoginRequiredMixin, CohortInstructorMixin, DetailView):
+    model = Graph
+    template_name = 'main/graph_clone_form.html'
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+        ctx.update({
+            'cohort': self.cohort,
+            'form': GraphCloneForm(user=self.request.user),
+        })
+        return ctx
+
+
+class GraphCloneFormView(LoginRequiredMixin, CohortInstructorMixin,
+                         SingleObjectMixin, FormView):
+    template_name = 'main/graph_clone_form.html'
+    form_class = GraphCloneForm
+    model = Graph
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.request.user
+        return kwargs
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+        ctx.update({
+            'cohort': self.cohort,
+            'form': GraphCloneForm(user=self.request.user),
+        })
+        return ctx
+
+    def post(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
```

Comment:
Isn't this handled through the mixin already?

Reply:
Correct... I left this in because it was in the docs I was following, and figured it couldn't hurt, but it's probably unnecessary. This is handled in the dispatch for LoginRequiredMixin and CohortInstructorMixin. I'll remove this.